presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClient.java

```diff
@@ -0,0 +1,453 @@
+import static io.prestosql.plugin.phoenix.PhoenixMetadata.DEFAULT_SCHEMA;
+import static io.prestosql.plugin.phoenix.TypeUtils.getArrayElementPhoenixTypeName;
+import static io.prestosql.plugin.phoenix.TypeUtils.getJdbcObjectArray;
+import static io.prestosql.plugin.phoenix.TypeUtils.jdbcObjectArrayToBlock;
+import static io.prestosql.plugin.phoenix.TypeUtils.toBoxedArray;
+import static io.prestosql.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.prestosql.spi.type.DoubleType.DOUBLE;
+import static io.prestosql.spi.type.RealType.REAL;
+import static io.prestosql.spi.type.TimeType.TIME;
+import static io.prestosql.spi.type.TimeWithTimeZoneType.TIME_WITH_TIME_ZONE;
+import static io.prestosql.spi.type.TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.sql.Types.ARRAY;
+import static java.sql.Types.FLOAT;
+import static java.sql.Types.LONGNVARCHAR;
+import static java.sql.Types.LONGVARCHAR;
+import static java.sql.Types.NVARCHAR;
+import static java.sql.Types.TIMESTAMP;
+import static java.sql.Types.TIMESTAMP_WITH_TIMEZONE;
+import static java.sql.Types.TIME_WITH_TIMEZONE;
+import static java.sql.Types.VARCHAR;
+import static java.util.Collections.nCopies;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+import static org.apache.phoenix.coprocessor.BaseScannerRegionObserver.SKIP_REGION_BOUNDARY_CHECK;
+import static org.apache.phoenix.query.QueryConstants.NULL_SCHEMA_NAME;
+import static org.apache.phoenix.util.SchemaUtil.ESCAPE_CHARACTER;
+
+public class PhoenixClient
+        extends BaseJdbcClient
+{
+    private final Configuration configuration;
+
+    @Inject
+    public PhoenixClient(PhoenixConfig config)
+            throws SQLException
+    {
+        super(
+                new BaseJdbcConfig(),
+                ESCAPE_CHARACTER,
+                new DriverConnectionFactory(
+                        DriverManager.getDriver(config.getConnectionUrl()),
+                        config.getConnectionUrl(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        config.getConnectionProperties()));
+        requireNonNull(config, "config is null");
+        configuration = new Configuration(false);
+        ConnectionInfo connectionInfo = PhoenixEmbeddedDriver.ConnectionInfo.create(config.getConnectionUrl());
+        connectionInfo.asProps()
+                .forEach(prop -> configuration.set(prop.getKey(), prop.getValue()));
+        config.getConnectionProperties()
+                .forEach((k, v) -> configuration.set((String) k, (String) v));
+    }
+
+    public PhoenixConnection getConnection(JdbcIdentity identity)
+            throws SQLException
+    {
+        Connection connection = connectionFactory.openConnection(identity);
```

Comment:
This try/catch seems to be wrong. I would assume that `openConnection()` is what will throw the error (not the `unwrap` call). Since `getConnection()` can throw `SQLException`, we should probably just remove the try/catch here and let the caller deal with it.

Reply:
There was concern in a prior review that in the (very) unlikely case that unwrap fails, the connection would never get closed.  Hence the try/catch here